pkg/syncer/metadatasyncer.go

```diff
@@ -631,20 +629,21 @@ func pvDeleted(obj interface{}, metadataSyncer *metadataSyncInformer) {
 		log.Warnf("PVDeleted: unrecognized object %+v", obj)
 		return
 	}
-	log.Debugf("PVDeleted: Deleting PV: %+v", pv)
 
-	// Verify if pv is a vsphere csi volume
-	if (pv.Spec.CSI == nil || pv.Spec.CSI.Driver != csitypes.Name) && (metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration) && pv.Spec.VsphereVolume == nil) {
-		log.Debugf("PVDeleted: Not a vSphere CSI Volume or migrated vSphere Volume: %+v", pv)
-		return
-	}
-
-	// Verify if pv is a vsphere volume and migration flag is set to false
-	if pv.Spec.VsphereVolume != nil && !metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration) {
-		log.Warnf("PVDeleted: volume-migration feature switch is disabled. Cannot delete vSphere volume metadata for %s", pv.Name)
-		return
+	migrationEnabled := metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration)
+	if migrationEnabled && pv.Spec.VsphereVolume != nil {
+		if !migratedToAnnotationExists(ctx, pv.Annotations) {
+			log.Info("migration annotation is not present on the PV. Skipping update")
```

Comment:
add "PVDeleted" in the beginning of log message

Reply:
Shouldn't the log message with "Skipping delete" ?

Reply:
done changed message to

```
			log.Debugf("pvDeleted: migration annotation is not present on the PV %q. Skipping deletion of PV metadata.", pv.Name)

```